File: 20 -- CoreList/CoreList/ViewController.m
//
//  ViewController.m
//  CoreList
//
//  Created by Jordan Anderson on 3/31/15.
//  Copyright (c) 2015 The Iron Yard. All rights reserved.
//

#import "ViewController.h"

@interface ViewController () <UITextFieldDelegate>

@property (weak, nonatomic) IBOutlet UITextField *addTaskTextField;
- (IBAction)saveItemButton:(UIButton *)sender;
- (IBAction)cancelItemButton:(UIButton *)sender;

@end

@implementation ViewController

- (void)viewDidLoad
{
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
}

- (void)didReceiveMemoryWarning
{
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (IBAction)saveItemButton:(UIButton *)sender
{
    if (![self.addTaskTextField.text isEqualToString:@""])
    {
        [self createNewItem];
        [self saveCoreDataUpdates];
        [self cancel];
    }
    
}


#pragma mark - TextField delegate

- (BOOL)textFieldShouldReturn:(UITextField *)textField
{
    BOOL rc = NO;
    
    if ([textField.text isEqualToString:@""])
    {
        rc = YES;
        
        [self.addTaskTextField becomeFirstResponder];
      
    }
    else if (![textField.text isEqualToString:@""])
    {
        [self createNewItem];
//        [self saveCoreDataUpdates];
        [self.addTaskTextField resignFirstResponder];
    }
    
    return rc;
}

- (void)createNewItem
{
   ListItem *aItem = [NSEntityDescription insertNewObjectForEntityForName:@"ListItem" inManagedObjectContext:self.cdStack.managedObjectContext]; // using core data / database to create a student object for us
    
    aItem.name = self.addTaskTextField.text;
}

- (void)saveCoreDataUpdates
{
    [self.cdStack saveOrFail:^(NSError *errorOrNil)
     {
         if (errorOrNil)
         {
             NSLog(@"Error from CDStack: %@", [errorOrNil localizedDescription]);
         }
     }];

}

- (IBAction)cancelItemButton:(UIButton *)sender
{
    [self cancel];
}

-(void)cancel
{
    [self dismissViewControllerAnimated:YES completion:nil];
}
@end

File: 21 -- MissionBriefingSwift/MissionBriefingSwift/ViewController.swift
//
//  ViewController.swift
//  MissionBriefingSwift
//
//  Created by Jordan Anderson on 4/1/15.
//  Copyright (c) 2015 The Iron Yard. All rights reserved.
//

import UIKit

class ViewController: UIViewController, UITextFieldDelegate
{
    @IBOutlet var agentNameTextField : UITextField!
    @IBOutlet var agentPasswordTextField : UITextField!
    @IBOutlet var greetingLabel : UILabel!
    @IBOutlet var missionBriefingTextView : UITextView!

    override func viewDidLoad()
    {
        super.viewDidLoad()
        agentNameTextField.text = ""
        greetingLabel.text = ""
        missionBriefingTextView.text = ""
        
        agentNameTextField.delegate = self
        agentPasswordTextField.delegate = self
    }

    override func didReceiveMemoryWarning()
    {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
    @IBAction func authenticateButton(sender : UIButton)
    {
        authenticateUser()
    }
    
    func textFieldShouldReturn(textField: UITextField!) -> Bool
    {
        var rc = true
            
            if textField == agentNameTextField
            {
             agentNameTextField.resignFirstResponder()
             agentPasswordTextField.becomeFirstResponder()
            }
            else if textField == agentPasswordTextField
            {
             agentPasswordTextField.resignFirstResponder()
             authenticateUser()
            }
        
        return rc;
    }
    
    func authenticateUser()
    {
        if !agentNameTextField.text.isEmpty && !agentPasswordTextField.text.isEmpty
        {
            var nameParts = agentNameTextField.text.componentsSeparatedByString(" ")
            var agentName = nameParts[0];
            if nameParts.count == 2
            {
                agentName = nameParts[1]
            }
            else if nameParts.count == 3
            {
                agentName = nameParts[2]
            }
            
            let greetingMessage = "Good evening, Agent \(agentName)"
            greetingLabel.text = greetingMessage;
            
            missionBriefingTextView.text = String(format:"This mission will be an arduous one, fraught with peril. You will cover much strange and unfamiliar territory. Should you choose to accept this mission, Agent \(agentName), you will certainly be disavowed, but you will be doing your country a great service. This message will self destruct in 5 seconds.")
            
            let swiftColor = UIColor(red: 0.585, green: 0.78, blue: 0.188, alpha: 1)
           self.view.backgroundColor = swiftColor
        }
        else
        {
            self.view.backgroundColor = UIColor.redColor()
        }
    }

}

